application/org.openjdk.jmc.flightrecorder.heatmap/src/main/resources/page.template

```diff
@@ -0,0 +1,16 @@
+<!DOCTYPE html "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd">
+<html xmlns="http://www.w3.org/1999/xhtml">
+  <head>
+    <meta http-equiv="content-type" content="text/html; charset=utf-8" />
+    <script>
+      %1$s
+    </script>
+  </head>
+
+  <body>
+    <div id="heatmap" width="100%%" style="text-align: center"></div>
```

Comment:
Extra "%" here ?

Reply:
That’s intentional 😊 We use `String.format` to build the HTML from this template file and we need to escape `%` so it doesn’t confuse it with the arguments.

This is one of the reasons I extracted the JS to a separate file, it was annoying to have to escape this everywhere (e.g. remainder operator, other percentages).

Reply:
Ah, neat! Thanks for the explanation!